platform/packages/platform/src/bank.rs

```diff
@@ -122,16 +123,24 @@ impl BankAccountView for BankView<'_> {
         Cmd: WithCoin<G> + Clone,
         Cmd::Output: Aggregate,
     {
-        self.querier
-            .query_all_balances(self.account)
-            .map_err(Error::CosmWasmQueryAllBalances)
-            .map(|cw_coins| {
-                cw_coins
-                    .into_iter()
-                    .filter_map(|cw_coin| maybe_from_cosmwasm_any::<G, _>(cw_coin, cmd.clone()))
-                    .reduce_results(Aggregate::aggregate)
+        G::currencies()
+            .filter_map(|ref currency| {
+                self.cw_balance(currency).map_err(Into::into).map_or_else(
+                    |err| Some(Err(err)),
+                    |ref cw_balance| {
+                        if cw_balance.amount == Uint128::zero() {
+                            None
+                        } else {
+                            Some(coin_legacy::from_cosmwasm_any::<G, _>(
+                                cw_balance,
+                                cmd.clone(),
+                            ))
+                        }
```

Comment:
The convenience method `bool::then` can be used for the purposes on conditionally constructing an option.
```suggestion
                        (cw_balance.amount != const { Uint128::zero() }).then(|| {
                            coin_legacy::from_cosmwasm_any::<G, _>(cw_balance, cmd.clone())
                        })
```

Reply:
thanks, did not recall to it